ui/app/AppLayouts/Chat/controls/community/CommunityPermissionsRow.qml

```diff
@@ -0,0 +1,315 @@
+                    model: holdingsListModel
+                }
+
+                StatusBaseText {
+                    visible: index !== (repeater.count - 1) || d.dotsVisible
+                    text: qsTr("or")
+                    font.pixelSize: root.pixelSize
+                    color: Theme.palette.baseColor1
+                }
+            }
+        }
+
+        StatusRoundedComponent {
+            Layout.preferredHeight: container.height
+            Layout.preferredWidth: Layout.preferredHeight
+
+            visible: d.dotsVisible
+            color: Theme.palette.baseColor3
+            border.color: root.backgroundColor
+            border.width: root.overlappingBorder
+
+            StatusIcon {
+                anchors.centerIn: parent
+                visible: d.dotsVisible
+                icon: "dots-icon"
+                height: root.dotsIconSize
+                width: height
+            }
+        }
+    }
+
+    onModelChanged: d.buildShortModel(root.model)
+    Component.onCompleted: d.buildShortModel(root.model)
+
+    ListModel{ id: shortModel }
+
+    component SinglePermissionRow :  RowLayout {
+        id: singlePermissionItem
+
+        readonly property int maxVisualTokens: 3
+
+        property var model
+        property string plusElementText: ""
+        property bool plusElementVisible: false
+
+        function buildTokensRowModel(model) {
+            shortTokensRowModel.clear()
+            if(!model)
+                return
+
+            var modelCount = model.rowCount()
+
+            // Need of shorter model
+            if(modelCount > maxVisualTokens) {
+                for(var i = 0; i < singlePermissionItem.maxVisualTokens - 1; i++)
+                    shortTokensRowModel.append(ModelUtils.get(model, i))
+
+                // Set Plus element properties:
+                plusElementVisible = true
+                if(modelCount === 4)
+                    plusElementText = qsTr("+2")
+
+                if(modelCount === 5)
+                    plusElementText = qsTr("+3")
+            } else {
+                // All elements in model
+                plusElementVisible = false
+                for(var j = 0; j < modelCount; j++)
+                    shortTokensRowModel.append(ModelUtils.get(model, j))
+            }
+        }
+
+        spacing: -root.overlapping
+
+        onModelChanged: buildTokensRowModel(singlePermissionItem.model)
+        Component.onCompleted: buildTokensRowModel(singlePermissionItem.model)
+
+        ListModel{ id: shortTokensRowModel }
+
+        Repeater {
+            model: HoldingsSelectionModel {
+                sourceModel: shortTokensRowModel
+
+                assetsModel: root.assetsModel
+                collectiblesModel: root.collectiblesModel
+            }
+
+            StatusRoundedImage {
+                Layout.preferredHeight: container.height
+                Layout.preferredWidth: Layout.preferredHeight
+
+                z: parent.z + index
```

Comment:
Imo it's correct, bc child could be located under the parent when has lower z. Please take a look at the forth example here: https://doc.qt.io/qt-5/qml-qtquick-item.html#z-prop.

Reply:
> @alexjba, imo it's correct, bc child could be located under the parent when has lower z. Please take a look at the forth example here: https://doc.qt.io/qt-5/qml-qtquick-item.html#z-prop.

This example will produce the same output as the example from documentation where the child is on top of the parent:

```
    Item {
        Rectangle {
            color: "red"
            width: 100; height: 100
            z: 200
            Rectangle {
                color: "blue"
                z: 0
                x: 50; y: 50; width: 100; height: 100
            }
        }
    }
```

The only way I know of to position it below the parent is to use negative z. In our case IMO it will be just an extra binding to the parent.z.

Reply:
ah, ok, there is no risk to go below parent. So I agree, parent.z could be skipped.

Reply:
Yeah, usually writing `z` explicitly means there's something fishy with the code; there are really very rare cases when you actually need it :)

Reply:
> Yeah, usually writing `z` explicitly means there's something fishy with the code; there are really very rare cases when you actually need it :)
Is it ok to use it here for the overlapping? If there is a better solution I'm open to it, this is the one I thought!